### ISSUE TITLE
Bug: UI unresponsive after click `run` with fsg/psg

### ISSUE THREAD

**Issue by mokurin000** [FHPythonUtils/Cli2Gui#22]
## Before You Begin

Before proceeding, please make sure to follow these steps:

- [x] I have checked for similar issues in the project's issue tracker.
- [x] I have searched closed issues to see if a similar problem was reported
  before.

## Issue Details

1. start cli2gui UI
2. enter arguments
3. click `Run`
4. UI will get stuck and warn unresponsive state 

### Description

This problem only occurs on pysimpleguiwrapper

### Expected Behavior

UI should not be blocked

### Actual Behavior

Unresponsive warn happens.

## System Information

Please provide the following additional information about your system or
environment:

- Operating System (OS): Windows 11
- OS Version: 22631.4249
- Browser (if applicable):
- Browser Version (if applicable):
- Project Version (if applicable): 2024.2

**Comment by FredHappyface** [comment]
Hey thanks for the issue, I wasn't able to replicate this locally. I've created a test file based on the code you'd shared with me previously (just removing most of the business type logic). Seems not to freeze up the ui.

![image](https://github.com/user-attachments/assets/3b464ded-0b4d-4e6a-b47a-d9826913d680)

**Comment by mokurin000** [comment]
> Hey thanks for the issue, I wasn't able to replicate this locally. I've created a test file based on the code you'd shared with me previously (just removing most of the business type logic). Seems not to freeze up the ui.
> 
> ![image](https://private-user-images.githubusercontent.com/41634689/373872421-3b464ded-0b4d-4e6a-b47a-d9826913d680.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgxMzMxMDIsIm5iZiI6MTcyODEzMjgwMiwicGF0aCI6Ii80MTYzNDY4OS8zNzM4NzI0MjEtM2I0NjRkZWQtMGI0ZC00ZTZhLWI0N2EtZDk4MjY5MTNkNjgwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDA1VDEyNTMyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRjYjE3ZDA5ODY3ZDQ4OGNlNjI4Y2Q2ZmM3OTk3NzU4NWExMjkzNjBjYTBkYTNmY2MzN2FkNDQxM2VhYWUzMmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.cAGAyJyNkI5vNVGaJ7pZ3ow-HsDK4o8Tbo4k3yAkigE)

could you please send your test script
I'm testing on https://github.com/mokurin000/fuckingfast-batch-download/commit/test-freeze

also, did you add some fake payload (for exmaple `time.sleep()` ?

**Comment by FredHappyface** [comment]
Yeah sure its https://github.com/FHPythonUtils/Cli2Gui/blob/master/tests/argparse/test_22.py

Can't seem to view the image 

Ah could it be the threading do you think? Shouldn't block dearpygui though? Unless I need to think about spawning a UI thread perhaps

**Comment by mokurin000** [comment]
> Yeah sure its https://github.com/FHPythonUtils/Cli2Gui/blob/master/tests/argparse/test_22.py
> 
> Can't seem to view the image
> 
> Ah could it be the threading do you think? Shouldn't block dearpygui though? Unless I need to think about spawning a UI thread perhaps

Ah, I think I may have misdescribe. I encountered blocking when I did not open another thread to start the main function. If Thread.start() is replaced with Thread.run(), it will run the task in the current thread, and then because the task is running in the UI thread, it will block the UI. What I did to solve the problem was to open the task in another thread (the wrapper)

**Comment by FredHappyface** [comment]
Ah right thanks for the clarification. I'll give that a go when I get the chance. I don't know enough about the inner workings of dearpygui to know if thread.run should block the UI thread but it sounds like it should? 

I'll do some investigation and get back to you if that's alright? :)

**Comment by mokurin000** [comment]
> Ah right thanks for the clarification. I'll give that a go when I get the chance. I don't know enough about the inner workings of dearpygui to know if thread.run should block the UI thread but it sounds like it should?
> 
> I'll do some investigation and get back to you if that's alright? :)

deerpygui

**Comment by mokurin000** [comment]
sorry for the misoperation..

dearpygui may spawns own event loop thread by default
While psg/fsg does not.
`Thread.run()` will simply run the `Callable` object in current thread
